spatialos-sdk/src/worker/alpha.rs

```diff
@@ -0,0 +1,465 @@
+    fn from_worker_sdk(response: &Worker_Alpha_PlayerIdentityTokenResponse) -> Self {
+        PlayerIdentityTokenResponse {
+            player_identity_token: cstr_to_string(response.player_identity_token),
+        }
+    }
+}
+
+pub struct PlayerIdentityTokenFuture {
+    internal: *mut Worker_Alpha_PlayerIdentityTokenResponseFuture,
+    consumed: bool,
+}
+
+impl PlayerIdentityTokenFuture {
+    fn new(ptr: *mut Worker_Alpha_PlayerIdentityTokenResponseFuture) -> Self {
+        PlayerIdentityTokenFuture {
+            internal: ptr,
+            consumed: false,
+        }
+    }
+
+    extern "C" fn callback_handler(
+        user_data: *mut ::std::os::raw::c_void,
+        response: *const Worker_Alpha_PlayerIdentityTokenResponse,
+    ) {
+        assert!(!response.is_null());
+        unsafe {
+            let response = *response;
+            let data =
+                &mut *(user_data as *mut Option<Result<PlayerIdentityTokenResponse, String>>);
+            if !response.error.is_null() {
+                let err = cstr_to_string(response.error);
+                *data = Some(Err(err));
+                return;
+            }
+
+            let response = PlayerIdentityTokenResponse::from_worker_sdk(&response);
+            *data = Some(Ok(response));
+        }
+    }
+
+    fn get_future(
+        &self,
+        timeout: Option<u32>,
+    ) -> Option<Result<PlayerIdentityTokenResponse, String>> {
+        assert!(!self.internal.is_null());
+
+        let mut data: Option<Result<PlayerIdentityTokenResponse, String>> = None;
+        unsafe {
+            Worker_Alpha_PlayerIdentityTokenResponseFuture_Get(
+                self.internal,
+                timeout.map_or(::std::ptr::null(), |value| &value),
+                &mut data as *mut _ as *mut ::std::os::raw::c_void,
+                Some(PlayerIdentityTokenFuture::callback_handler),
+            );
+        }
+
+        data
+    }
+}
+
+impl Future for PlayerIdentityTokenFuture {
+    type Item = PlayerIdentityTokenResponse;
+    type Error = String;
+
+    fn poll(&mut self) -> Result<Async<<Self as Future>::Item>, <Self as Future>::Error> {
+        if self.consumed {
+            return Err("PlayerIdentityTokenFuture has already been consumed".to_owned());
+        }
+
+        self.get_future(Some(0))
+            .map_or(Ok(Async::NotReady), |result| {
+                self.consumed = true;
+                result.map(Async::Ready)
+            })
+    }
+
+    fn wait(self) -> Result<<Self as Future>::Item, <Self as Future>::Error>
+        where
+            Self: Sized,
+    {
+        if self.consumed {
+            return Err("PlayerIdentityTokenFuture has already been consumed".to_owned());
+        }
+
+        self.get_future(None).expect("Blocking call to Worker_Alpha_PlayerIdentityTokenResponseFuture_Get did not trigger callback")
+    }
+}
+
+impl Drop for PlayerIdentityTokenFuture {
+    fn drop(&mut self) {
+        if !self.internal.is_null() {
```

Comment:
Same question here: Does it make sense to do the null check here, or would it be better to assert that the pointer isn't null in `PlayerIdentityTokenFuture::new`?